src/sagemaker/jumpstart/types.py

```diff
@@ -940,10 +1004,340 @@ def to_json(self) -> Dict[str, Any]:
+    ) -> JumpStartMetadataConfig:
+        """Gets the best the config based on config ranking."""
+        if self.configs and (
+            not self.config_rankings or not self.config_rankings.get(ranking_name)
+        ):
+            raise ValueError("Config exists but missing config ranking.")
+
+        if self.scope == JumpStartScriptScope.INFERENCE:
+            instance_type_attribute = "supported_inference_instance_types"
+        elif self.scope == JumpStartScriptScope.TRAINING:
+            instance_type_attribute = "supported_training_instance_types"
+        else:
+            raise ValueError(f"Unknown script scope {self.scope}")
+
+        rankings = self.config_rankings.get(ranking_name)
+        for config_name in rankings.rankings:
+            resolved_config = self.configs[config_name].resolved_config
+            if instance_type and instance_type not in getattr(
+                resolved_config, instance_type_attribute
+            ):
+                continue
+            return self.configs[config_name]
+
+        return None
+
+
+class JumpStartModelSpecs(JumpStartMetadataBaseFields):
+    """Data class JumpStart model specs."""
+
+    slots = [
+        "inference_configs",
+        "inference_config_components",
+        "inference_config_rankings",
+        "training_configs",
+        "training_config_components",
+        "training_config_rankings",
+    ]
+
+    __slots__ = JumpStartMetadataBaseFields.__slots__ + slots
+
+    def __init__(self, spec: Dict[str, Any]):  # pylint: disable=super-init-not-called
+        """Initializes a JumpStartModelSpecs object from its json representation.
+
+        Args:
+            spec (Dict[str, Any]): Dictionary representation of spec.
+        """
+        self.from_json(spec)
+        if self.inference_configs and self.inference_configs.get_top_config_from_ranking():
+            super().from_json(self.inference_configs.get_top_config_from_ranking().resolved_config)
+
+    def from_json(self, json_obj: Dict[str, Any]) -> None:
+        """Sets fields in object based on json of header.
+
+        Args:
+            json_obj (Dict[str, Any]): Dictionary representation of spec.
+        """
+        super().from_json(json_obj)
+        self.inference_config_components: Optional[Dict[str, JumpStartConfigComponent]] = (
+            {
+                component_name: JumpStartConfigComponent(component_name, component)
+                for component_name, component in json_obj["inference_config_components"].items()
+            }
+            if json_obj.get("inference_config_components")
+            else None
+        )
+        self.inference_config_rankings: Optional[Dict[str, JumpStartConfigRanking]] = (
+            {
+                alias: JumpStartConfigRanking(ranking)
+                for alias, ranking in json_obj["inference_config_rankings"].items()
+            }
+            if json_obj.get("inference_config_rankings")
+            else None
+        )
```

Comment:
optional: inlining dict comprehension with ternary operator makes the code a bit hard to read IMHO, consider:

```py
if json_obj.get("inference_config_rankings"):
  self.inference_config_rankings = { ... }
else:
  self.inference_config_rankings = None
```